pandas/core/indexers.py

```diff
@@ -376,6 +376,25 @@ def unpack_1tuple(tup):
     return tup
 
 
+def check_key_length(columns, key, value):
+    """Checks if a key used as indexer has the same length as the columns it is
+    associated with.
+
+    Parameters
+    ----------
+    columns: The columns of the DataFrame to index.
+    key: A list-like of keys to index with.
+    value: The value to set for the keys.
+
+    """
+    if columns.is_unique:
+        if len(value.columns) != len(key):
+            raise ValueError("Columns must be same length as key")
+    else:
+        if len(columns.get_indexer_non_unique(key)[0]) != len(value.columns):
```

Comment:
do we necessarily have `all(col in columns for col in key)`?

Reply:
No, but this would return -1 as indexer for this col. I should probably add a note about this :)